packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -110,23 +128,34 @@ export function makeRumPublicApi<C extends RumInitConfiguration>(startRumImpl: S
     isAlreadyInitialized = true
 
     function doStartRum(initialViewName?: string) {
```

Comment:
What about moving doStartRum out of initRum as it starts to become bigger and bigger. 
Not directly related to this PR but, IMO, the use of closures with 3 levels of depth `makeRumPublicApi` -> `initRum` -> `doStartRum`, is hard to read.